admiral/camp_functions.sqf

```diff
@@ -275,73 +271,79 @@ adm_camp_fnc_getSpawnFunction = {
-    FUN_ARGS_1(_zone);
+    params ["_zone"];
 
     [_zone, GROUP_TYPE_INF, UNIT_TYPE_INF, adm_camp_fnc_placeMan] call adm_camp_fnc_periodicSpawnGroups;
 };
 
 adm_camp_fnc_periodicSpawnTechGroups = {
-    FUN_ARGS_1(_zone);
+    params ["_zone"];
 
     [_zone, GROUP_TYPE_TECH, UNIT_TYPE_INF, adm_camp_fnc_placeMan] call adm_camp_fnc_periodicSpawnGroups;
 };
 
 adm_camp_fnc_periodicSpawnArmourGroups = {
-    FUN_ARGS_1(_zone);
+    params ["_zone"];
 
     [_zone, GROUP_TYPE_ARMOUR, UNIT_TYPE_CREW, adm_camp_fnc_placeMan] call adm_camp_fnc_periodicSpawnGroups;
 };
 
 adm_camp_fnc_periodicSpawn = {
-    FUN_ARGS_1(_zone);
+    params ["_zone"];
 
-    private ["_zoneInfGroups", "_zoneTechGroups", "_zoneArmourGroups"];
-    _zoneInfGroups = GET_ZONE_SPAWNED_GROUPS(_zone) select 0;
-    _zoneTechGroups = GET_ZONE_SPAWNED_GROUPS(_zone) select 1;
-    _zoneArmourGroups = GET_ZONE_SPAWNED_GROUPS(_zone) select 2;
+    private _zoneInfGroups = GET_ZONE_SPAWNED_GROUPS(_zone) select 0;
+    private _zoneTechGroups = GET_ZONE_SPAWNED_GROUPS(_zone) select 1;
+    private _zoneArmourGroups = GET_ZONE_SPAWNED_GROUPS(_zone) select 2;
     SET_CAMP_LAST_SPAWN_TIME(_zone,AS_ARRAY_3(diag_tickTime,diag_tickTime,diag_tickTime));
-    waitUntil {
-        DECLARE(_spawnedGroups) = [];
-        _spawnedGroups = [_zone, GROUP_TYPE_INF, adm_camp_fnc_periodicCanSpawnGroups, adm_camp_fnc_periodicSpawnInfGroups] call adm_camp_fnc_trySpawnGroups;
-        _zoneInfGroups append _spawnedGroups;
-        INFO("admiral.camp",FMT_2("Periodic Camp Zone '%1' spawned '%2' infantry group(s).",GET_ZONE_ID(_zone),count _spawnedGroups));
-        _spawnedGroups = [];
-        _spawnedGroups = [_zone, GROUP_TYPE_TECH, adm_camp_fnc_periodicCanSpawnGroups, adm_camp_fnc_periodicSpawnTechGroups] call adm_camp_fnc_trySpawnGroups;
-        _zoneTechGroups append _spawnedGroups;
-        INFO("admiral.camp",FMT_2("Periodic Camp Zone '%1' spawned '%2' technical group(s).",GET_ZONE_ID(_zone),count _spawnedGroups));
-        _spawnedGroups = [];
-        _spawnedGroups = [_zone, GROUP_TYPE_ARMOUR, adm_camp_fnc_periodicCanSpawnGroups, adm_camp_fnc_periodicSpawnArmourGroups] call adm_camp_fnc_trySpawnGroups;
-        _zoneArmourGroups append _spawnedGroups;
-        INFO("admiral.camp",FMT_2("Periodic Camp Zone '%1' spawned '%2' armour group(s).",GET_ZONE_ID(_zone),count _spawnedGroups));
-        ["camp.spawned.groups", [_zoneInfGroups, _zoneTechGroups, _zoneArmourGroups, _zone]] call adm_event_fnc_emitEvent;
-
-        sleep GET_CAMP_DELAY(_zone);
-        [_zone] call adm_camp_fnc_isPoolEmpty || {!IS_CAMP_ENABLED(_zone)};
-    };
-    INFO("admiral.camp",FMT_1("Periodic Camp Zone '%1' has stopped.",GET_ZONE_ID(_zone)));
+
+    [
+        {
+            params ["_args", "_id"];
+            _args params ["_zone","_zoneInfGroups","_zoneTechGroups","_zoneArmourGroups"];
+
+            if ([_zone] call adm_camp_fnc_isPoolEmpty || {!IS_CAMP_ENABLED(_zone)}) exitWith {
```

Comment:
This should be the last statement as in the `waitUntil` check, so the spawning runs at least once.

Reply:
Moved this for all 3 of the camp types to be post at least one run of the code.